src/components/EditableComponent.tsx

```diff
@@ -19,12 +19,14 @@ import { ContainerState } from './Container';
 /**
  * Configuration object of the withEditable function.
  *
- * @property emptyLabel Label to be displayed on the overlay when the component is empty.
- * @property isEmpty Callback function to determine if the component is empty.
+ * @property emptyLabel - Label to be displayed on the overlay when the component is empty
+ * @property isEmpty - Callback function to determine if the component is empty
+ * @property aemResourceType - AEM ResourceType to be added as an attribute on the editable component dom
  */
 export interface EditConfig<P extends MappedComponentProperties> {
     emptyLabel?: string;
     isEmpty(props: P): boolean;
+    aemResourceType?: string;
```

Comment:
As discussed `resourceType` should be enough as the object is already meant for configuring the editability of the component in AEM